dotCMS/src/main/java/com/dotmarketing/common/reindex/IndexResource.java

```diff
@@ -0,0 +1,437 @@
+        
+        
+        
+        
+        message=message.replace("{0}", String.valueOf(data.get("successfulShards")));
+        message=message.replace("{1}", String.valueOf(data.get("failedShards")));
+        
+        
+        sendAdminMessage(message, MessageSeverity.INFO, init.getUser(),5000);
+        
+        
+        
+        return Response.ok(new ResponseEntityView(data)).build();
+
+    }
+
+    private InitDataObject validateUser(final HttpServletRequest request, final HttpServletResponse response) {
+        return new WebResource
+                        .InitBuilder(request, response)
+                        .requiredRoles(Role.CMS_ADMINISTRATOR_ROLE)
+                        .requiredPortlet("maintenance")
+                        .init();
+
+    }
+
+    @CloseDBIfOpened
+    @GET
+    @JSONP
+    @NoCache
+    @Path("/cluster")
+    @Produces({MediaType.APPLICATION_JSON})
+    public Response getClusterStats(@Context final HttpServletRequest request, @Context final HttpServletResponse response)
+                    throws DotDataException {
+
+        validateUser(request, response);
+
+
+        final ESIndexAPI esIndexAPI = new ESIndexAPI();
+        final ClusterStats clusterStats = esIndexAPI.getClusterStats();
+
+        Builder<String, Object> builder =
+                        ImmutableMap.<String, Object>builder().put("clusterName", clusterStats.getClusterName());
+
+        for (NodeStats stats : clusterStats.getNodeStats()) {
+            builder.put("name", stats.getName())
+                .put("master", stats.isMaster())
+                .put("host", stats.getHost())
+                .put("address", stats.getTransportAddress())
+                .put("size", stats.getSize())
+                .put("count", stats.getDocCount());
+        }
+        return Response.ok(new ResponseEntityView(builder.build())).build();
+    }
+    
+    @CloseDBIfOpened
+    @GET
+    @JSONP
+    @NoCache
+    @Path("/failed")
+    @Produces({MediaType.APPLICATION_JSON})
+    public Response downloadRemainingRecordsAsCsv(@Context final HttpServletRequest request,
+                    @Context final HttpServletResponse response) throws DotDataException {
+        validateUser(request, response);
+        List<Map<String, Object>> results = new ArrayList<>();
+        
+        
+        
+        APILocator.getReindexQueueAPI().getFailedReindexRecords().stream().forEach(row->{
+            Map<String, Object> failure = new HashMap<>();
+            
+            final Contentlet contentlet = Try.of(()->APILocator.getContentletAPI().findContentletByIdentifierAnyLanguage(row.getIdentToIndex()))
+                            .onFailure(e->Logger.warn(IndexResource.class, e.getMessage(), e))
+                            .getOrElse(Contentlet::new);
+            Map<String, Object> conMap = Try.of(() -> new ContentletToMapTransformer(contentlet).toMaps().get(0))
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Local variable 'conMap' could be declared final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5632525)